File: tests/machine/x/php/update_stmt.php
<?php
class Person {
    public $name;
    public $age;
    public $status;
    public function __construct($fields = []) {
        $this->name = $fields['name'] ?? null;
        $this->age = $fields['age'] ?? null;
        $this->status = $fields['status'] ?? null;
    }
}
$people = [
    new Person([
        'name' => "Alice",
        'age' => 17,
        'status' => "minor"
    ]),
    new Person([
        'name' => "Bob",
        'age' => 25,
        'status' => "unknown"
    ]),
    new Person([
        'name' => "Charlie",
        'age' => 18,
        'status' => "unknown"
    ]),
    new Person([
        'name' => "Diana",
        'age' => 16,
        'status' => "minor"
    ])
];
foreach ($people as $_tmp0 => $_tmp1) {
    $name = $_tmp1->name;
    $age = $_tmp1->age;
    $status = $_tmp1->status;
    if ($age >= 18) {
        $_tmp1->status = "adult";
        $_tmp1->age = $age + 1;
    }
    $people[$_tmp0] = $_tmp1;
}
echo "ok", PHP_EOL;
?>

File: tests/human/x/pas/update_stmt.pas
program UpdateStmt;

type
  Person = record
    name: string;
    age: integer;
    status: string;
  end;

var
  people: array[1..4] of Person = (
    (name:'Alice'; age:17; status:'minor'),
    (name:'Bob'; age:25; status:'unknown'),
    (name:'Charlie'; age:18; status:'unknown'),
    (name:'Diana'; age:16; status:'minor')
  );
  i: integer;
begin
  for i := 1 to Length(people) do
    if people[i].age >= 18 then
    begin
      people[i].status := 'adult';
      people[i].age := people[i].age + 1;
    end;

  Writeln('ok');
end.
